File: backend/app/schemas/daily_summary.py
from pydantic import BaseModel
from datetime import datetime, date
from typing import List


class DailySummaryStatusUpdate(BaseModel):
    id: int
    title: str
    author: str
    created_at: datetime


class DailySummaryIncident(BaseModel):
    id: int
    title: str
    severity: str
    status: str


class DailySummaryBlocker(BaseModel):
    id: int
    description: str
    status: str


class DailySummaryDecision(BaseModel):
    id: int
    title: str
    decision_date: date


class DailySummaryStatistics(BaseModel):
    total_status_updates: int
    critical_incidents: int
    active_blockers: int
    decisions_last_7_days: int


class DailySummaryContent(BaseModel):
    status_updates: List[DailySummaryStatusUpdate]
    incidents: List[DailySummaryIncident]
    blockers: List[DailySummaryBlocker]
    recent_decisions: List[DailySummaryDecision]
    statistics: DailySummaryStatistics


class DailySummaryListItem(BaseModel):
    id: int
    summary_date: date
    status_updates_count: int
    incidents_count: int
    blockers_count: int
    decisions_count: int
    generated_at: datetime

    class Config:
        from_attributes = True


class DailySummary(BaseModel):
    id: int
    summary_date: date
    content: DailySummaryContent
    status_updates_count: int
    incidents_count: int
    blockers_count: int
    decisions_count: int
    generated_at: datetime

    class Config:
        from_attributes = True


class DailySummaryList(BaseModel):
    items: List[DailySummaryListItem]
    total: int
    page: int
    limit: int

File: frontend/src/types/dailySummary.ts
export interface DailySummaryStatusUpdate {
  id: number
  title: string
  author: string
  created_at: string
}

export interface DailySummaryIncident {
  id: number
  title: string
  severity: string
  status: string
}

export interface DailySummaryBlocker {
  id: number
  description: string
  status: string
}

export interface DailySummaryDecision {
  id: number
  title: string
  decision_date: string
}

export interface DailySummaryStatistics {
  total_status_updates: number
  critical_incidents: number
  active_blockers: number
  decisions_last_7_days: number
}

export interface DailySummaryContent {
  status_updates: DailySummaryStatusUpdate[]
  incidents: DailySummaryIncident[]
  blockers: DailySummaryBlocker[]
  recent_decisions: DailySummaryDecision[]
  statistics: DailySummaryStatistics
}

export interface DailySummaryListItem {
  id: number
  summary_date: string
  status_updates_count: number
  incidents_count: number
  blockers_count: number
  decisions_count: number
  generated_at: string
}

export interface DailySummary {
  id: number
  summary_date: string
  content: DailySummaryContent
  status_updates_count: number
  incidents_count: number
  blockers_count: number
  decisions_count: number
  generated_at: string
}

export interface DailySummaryList {
  items: DailySummaryListItem[]
  total: number
  page: number
  limit: number
}
